pytrip/utils/slicedos.py

```diff
@@ -0,0 +1,213 @@
+from pytrip import dos
+from pytrip import ctx
+import argparse
+import logging
+from numpy import arange, meshgrid, ma
+import matplotlib.pyplot as plt
+from matplotlib import colors
+import os
+import sys
+
+logger = logging.getLogger(__name__)
+
+
+def check_compatible(a, b):
+    """ Simple comparison of cubes. if X,Y,Z dims are the same, and
+    pixel sizes as well, then they are compatible. (Duck typed)
+    """
+    eps = 1e-5
+
+    if a.dimx != b.dimx:
+        logger.error("DIMX does not match: "+str(a.dimx)+" "+str(b.dimx))
+        raise Exception("Cubes don't match, check dimx in header.")
+
+    if a.dimy != b.dimy:
+        logger.error("DIMY does not match: "+str(a.dimy)+" "+str(b.dimy))
+        raise Exception("Cubes don't match, check dimy in header.")
+
+    if a.dimz != b.dimz:
+        logger.error("DIMZ does not match: "+str(a.dimz)+" "+str(b.dimz))
+        raise Exception("Cubes don't match, check dimz in header.")
+
+    if (a.pixel_size - b.pixel_size) > eps:
+        logger.error("Pixel size does not match: "+str(a.pixel_size)+" "+str(b.pixel_size))
+        raise Exception("Cubes don't match, check pixel_size in header.")
+
+    if a.slice_dimension != b.slice_dimension:
+        logger.error("Slice dimension does not match: "+str(a.slice_dimension)+" "+str(b.slice_dimension))
+        raise Exception("Cubes don't match, check slice_dimension in header.")
+    return True
+
+
+def main(args):
+    if args.verbosity == 1:
+        logger.basicConfig(level=logging.INFO)
+    if args.verbosity > 1:
+        logger.basicConfig(level=logging.DEBUG)
+
+    logger.info("Dos file: " + args.dos)
+
+    dosbasename = args.dos.split(".")[-2]
+    ctxbasename = args.ctx.split(".")[-2]
+
+    d = dos.DosCube()
+    fndos = dosbasename + ".dos"
```

Comment:
For a future reference: ".dos" can be obtained from a `data_file_extension` field in `DosCube` class.